app/src/composables/use-relation-multiple.ts

```diff
@@ -440,13 +440,29 @@ export function useRelationMultiple(
 		watch(
 			selectedOnPage,
 			(newVal, oldVal) => {
-				if (!isEqual(newVal, oldVal)) loadSelectedDisplay();
+				if (newVal.length !== oldVal?.length || !isEqual(newVal.map(getRelatedIDs), (oldVal ?? []).map(getRelatedIDs)))
+					loadSelectedDisplay();
 			},
 			{ immediate: true }
 		);
 
 		return { fetchedSelectItems, selected, isItemSelected, selectedOnPage };
 
+		function getRelatedIDs(item: DisplayItem) {
```

Comment:
should we add the return type to this function? so it would be 
```javascript
function getRelatedIDs(item: DisplayItem) : string | undefined { 
     ...
 }
```
this would make it more clear on how to use the function and it might prevent future bugs